webpack.config.js

```diff
@@ -21,21 +27,59 @@ module.exports = (env) => {
       /\bwebpack\/buildin\b/
     ];
 
-    filename += '.bundle';
+    babelOptions = {
+      sourceType: 'unambiguous',
+      ignore: [/\/core-js/],
+      presets: [
+        ['@babel/preset-env',
+          {
+            debug: false, /* set to `true` if you start banging your head against the wall */
+            targets: [
+              'last 2 versions',
+              '> 5%',
+              'IE 11',
+              'not dead'
+            ]
+          }
+        ]
+      ],
+      plugins: [
+        ['@babel/plugin-transform-runtime', {
+          corejs: 3
+        }]
+      ]
+    };
   } else {
-    /* if we are building for 'module', don't polyfill/transpile any dependencies */
-    babelExcludes = [/node_modules/];
+    /* we are building for node */
+    const modulesToBundle = [/* left this here just in case it is needed */];
+
+    /* we don't want to bundle node_modules */
+    externals.push(nodeExternals({
+      allowlist: modulesToBundle.map(m => new RegExp(`^${m.replace(/-/g, '\\-')}`))
+    }));
+
+    /* if we are building for 'module', don't polyfill/transpile most dependencies */
+    babelExcludes = [
+      new RegExp(`/node_modules/(?!${modulesToBundle.join('|')}/)`)
+    ];
+
+    babelOptions = {
+      sourceType: 'unambiguous',
+      presets: ['@babel/preset-env']
+    };
   }
 
   filename += minimize ? '.min.js' : '.js';
+  mode = minimize ? 'production' : 'development';
 
   return {
-    target: cdn ? 'web' : 'node',
+    target: 'web',
```

Comment:
After a ton of trial and error, this was one of the main pieces I was missing. Even though we are building to a "bundled/cdn" version _and_ a "node/commonjs" version, we still need to target `'web'`. This will allow webpack to use 3rd party libs' `package.json#browser` field. It causes a ton of build headaches for consumers of the streaming-client if the target is `'node'` (because the builders can't find `crypto`, `node-stringprep`, and a few others).